src/pages/ProductDetailsPage/ProductDetailsPage.tsx

```diff
@@ -1,9 +1,229 @@
 'use strict';
+import cn from 'classnames';
+import { Link } from 'react-router-dom';
+import { useEffect, useState } from 'react';
+import { useParams } from 'react-router-dom';
+import { client } from '../../utils/http/fetchClient';
+import { ProductDetails } from '../../types/ProductDetails';
+import style from './ProductDetailsPage.module.scss';
 
 export const ProductDetailsPage = () => {
+  const { productId } = useParams();
+
+  const [product, setProduct] = useState<ProductDetails | null>(null);
+  const [mainImage, setMainImage] = useState(product?.images[0]);
+  const [activeColor, setActiveColor] = useState<string | null>(null);
+  const [activeMemory, setActiveMemory] = useState<string | null>(null);
+  const [isLoading, setIsLoading] = useState(true);
+  const [isError, setIsError] = useState(false);
+
+  useEffect(() => {
+    setIsError(false);
+
+    client.get<ProductDetails>(`/products/${productId}`)
+      .then((data) => {
+        setProduct(data);
+        setMainImage(data.images[0]);
+        setActiveColor(data.color);
+        setActiveMemory(data.capacity.toLocaleLowerCase());
+      })
+      .catch(() => setIsError(true))
+      .finally(() => setIsLoading(false));
+  },[productId]);
+
+  console.log(activeColor, activeMemory);
+  const descriptions = product?.description;
+
+  const specs = [
+    { name: 'Screen', value: product?.screen },
+    { name: 'Resolution', value: product?.resolution },
+    { name: 'Processor', value: product?.processor },
+    { name: 'RAM', value: product?.ram },
+    { name: 'Built in memory', value: product?.capacity },
+    { name: 'Camera', value: product?.camera },
+    { name: 'Zoom', value: product?.zoom },
+    { name: 'Cell', value: product?.cell.join(', ') },
+  ];
+
   return (
     <>
-      <h1>ProductDetailsPage</h1>
+      {isLoading && <h1>Loading</h1>}
+      {isError && <h1>Error</h1>}
+      {(product && !isError && !isLoading) && (
+        <section className={style.itemCard}>
+          <h1 className={style.mainTitle}>{product?.name}</h1>
+
+          <article className={style.gallery}>
+            <div className={style.galleryBlock}>
+              <div className={style.galleryBlock__leftSide}>
+
+                {product?.images.map((image) => (
+                  <img
+                    src={image}
+                    key={image}
+                    alt='iphone'
+                    style={mainImage === image ? { border: '2px solid #000'}: {}}
+                    onClick={(event) => {
+                      const target = event.target as HTMLImageElement;
+                      setMainImage(target.src);
+                    }}
+                  />
+                ))}
+
+              </div>
+              <div className={style.galleryBlock__rightSide}>
+                <img src={mainImage} alt='iphone' />
+              </div>
+            </div>
+
+            <div className={style.configure}>
```

Comment:
Here it is better to put it in component (components/FeaturesDetails) 

Such data types are not called configure, it is called features